pkg/operator/ceph/object/controller.go

```diff
@@ -368,22 +368,20 @@ func (r *ReconcileCephObjectStore) reconcileCreateObjectStore(cephObjectStore *c
 	if cephObjectStore.Spec.IsExternal() {
 		logger.Info("reconciling external object store")
 
-		// RECONCILE SERVICE
-		logger.Info("reconciling object store service")
-		err = cfg.reconcileService(cephObjectStore)
-		if err != nil {
-			return r.setFailedStatus(k8sutil.ObservedGenerationNotAvailable, namespacedName, "failed to reconcile service", err)
-		}
+		// TODO: remove this code block for v1.13 release
+		// upgrade case: remove previously-existing service
+		service := cfg.generateService(cephObjectStore)
+		// TODO: list ObjectBuckets to see if any reference this service; if so, skip deleting it
 
-		// RECONCILE ENDPOINTS
-		// Always add the endpoint AFTER the service otherwise it will get overridden
-		logger.Info("reconciling external object store endpoint")
-		err = cfg.reconcileExternalEndpoint(cephObjectStore)
+		// deleting the service also deletes the endpoint(s), so deleting them isn't necessary
+		err := k8sutil.DeleteService(cfg.clusterInfo.Context, cfg.context.Clientset, service.Namespace, service.Name)
 		if err != nil {
-			return r.setFailedStatus(k8sutil.ObservedGenerationNotAvailable, namespacedName, "failed to reconcile external endpoint", err)
+			logger.Warningf("continuing after non critical error. "+
```

Comment:
If the service doesn't exist we don't need to log this, right?

Reply:
If it doesn't exist, there will be no error from the called function. This is in case of some other error.